### PR TITLE
Add Download for Teams and Block CSV

### PR DESCRIPTION
**Pull request by justinkonecny**
Add admin-only button to the `available-teams` page that downloads a CSV of all Team/User data

### CODE REVIEW

File: src/constants/userConstants.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+const STANDARD = 'STANDARD';
+const ADMIN = 'ADMIN';
+
+export default {
+  STANDARD,
+  ADMIN,
+};
```
Comment (florisdobber) [inline src/constants/userConstants.js:7]: These are already defined in auth/constants

File: src/views/AvailableTeams.vue [hunk 2]
```diff
@@ -29,21 +29,34 @@
...
+      </b-button>
   </div>
 </template>
 
 <script>
 import Vue from 'vue';
 import VueRouter from 'vue-router';
 import { mapState } from 'vuex';
+import { getTeamsCSV } from '../api/api';
+import userConstants from '../constants/userConstants';
```
Comment (florisdobber) [inline src/views/AvailableTeams.vue:45]: So you can change this import to from '../auth/constants'

File: src/constants/userConstants.js (hunk 1 above)
Comment (justinkonecny) [reply src/constants/userConstants.js:7]: Ah, thank you @florisdobber, somehow I missed these when looking.

@jackblanc the backend returns something like:
```
{
    "firstName": "Justin",
    "lastName": "Konecny",
    "username": "jkonecny",
    "email": "justin@gmail.com",
    "privilegeLevel": "ADMIN"
}
```

Comment (justinkonecny) [reply src/constants/userConstants.js:7]: @florisdobber thanks--I don't know how I missed that.
@jackblanc yep yep, my mistake, I'm fixing that now!

File: src/views/AvailableTeams.vue (hunk 2 above)
Comment (justinkonecny) [reply src/views/AvailableTeams.vue]: Fixed, see 877e07032f295e2ca1f3cd244d365807356b4b5a.

File: src/views/AvailableTeams.vue [hunk 3]
```diff
@@ -57,6 +70,31 @@ export default {
     createTeam() {
       this.$router.push('/create');
     },
+    /**
+     * Downloads a CSV that contains all Team/User information.
+     */
+    async downloadCSV() {
+      try {
+        const resp = await getTeamsCSV();
```
[the remark this answers is not in the document]
Comment (justinkonecny) [reply src/views/AvailableTeams.vue]: Fixed, see 877e07032f295e2ca1f3cd244d365807356b4b5a.

**Comment by justinkonecny** [comment]
Screen Shot 2020-06-11 at 2 55 07 PM
Produces something like this.

File: src/views/AvailableTeams.vue [hunk 4]
```diff
@@ -70,18 +79,17 @@ export default {
...
-    async downloadCSV() {
-      try {
-        const resp = await getTeamsCSV();
-        if (resp.status === 200) {
-          this.forceFileDownload(resp.data, 'Teams Export Data');
-        }
-      } catch (error) {
-        // Something went wrong
-      }
+    downloadTeamsCSV() {
```
Comment (jackblanc) [inline src/views/AvailableTeams.vue:91]: Thanks 👍

**Pull request merged**